llvm/include/llvm/IR/Type.h

```diff
@@ -165,6 +168,15 @@ class Type {
   /// Return true if this is powerpc long double.
   bool isPPC_FP128Ty() const { return getTypeID() == PPC_FP128TyID; }
 
+  /// Return true if this is 'decimal32'.
+  bool isDecimal32Ty() const { return getTypeID() == Decimal32TyID; }
+
+  /// Return true if this is 'decimal64'.
+  bool isDecimal64Ty() const { return getTypeID() == Decimal64TyID; }
+
+  /// Return true if this is 'decimal128'.
+  bool isDecimal128Ty() const { return getTypeID() == Decimal128TyID; }
+
```

Comment:
I think you also need a general `isDecimalFPTy()` helper method too, to be able to distinguish between binary and decimal floating point type.

Reply:
Oh! That's the response to my question above.